libsolidity/formal/SMTEncoder.cpp

```diff
@@ -779,7 +779,16 @@ void SMTEncoder::visitTypeConversion(FunctionCall const& _funCall)
 	auto argument = _funCall.arguments().front();
 	unsigned argSize = argument->annotation().type->storageBytes();
 	unsigned castSize = _funCall.annotation().type->storageBytes();
-	if (argSize == castSize)
+	// Allow same-size casts as well as literal arguments.
+	// TODO: support StringLiteral
+	if (auto category = argument->annotation().type->category(); category == Type::Category::RationalNumber)
+	{
+		auto const& numberType = dynamic_cast<RationalNumberType const&>(*argument->annotation().type);
+		// TODO: while we can supply the literal here, it is actually not used in RationalNumberType::literalValue()
+		// defineExpr(_funCall, numberType.literalValue(dynamic_cast<Literal const*>(argument.get())));
+		defineExpr(_funCall, numberType.literalValue(nullptr));
```

Comment:
I think we should also add the type constraints for `_funCall`, since right now I'm not sure any code is doing that. You can call function `setSymbolicUnknownValue` which will do that.

Reply:
Ah no, that's not needed, sorry.

Reply:
Does this work for sign conversion? `uint(-1)`?

Reply:
There is a test for `uint(-1)`.